packages/backend/src/models/GroupsModel.ts

```diff
@@ -187,38 +189,83 @@ export class GroupsModel {
         return deletedRows.length > 0;
     }
 
-    async updateGroup(groupUuid: string, update: UpdateGroup): Promise<Group> {
-        // Updates with joins not supported by knex
-        const results = await this.database.raw<{
-            rows: {
-                group_uuid: string;
-                name: string;
-                created_at: Date;
-                organization_uuid: string;
-            }[];
-        }>(
-            `
-            UPDATE groups
-            SET name = :name
-            FROM organizations
-            WHERE groups.group_uuid = :groupUuid
-            AND groups.organization_id = organizations.organization_id
-            RETURNING groups.group_uuid, groups.name, groups.created_at, organizations.organization_uuid
-            `,
-            {
-                groupUuid,
-                name: update.name,
-            },
-        );
-        const [updatedGroup] = results.rows;
-        if (updatedGroup === undefined) {
+    async updateGroup(
+        groupUuid: string,
+        update: UpdateGroupWithMembers,
+    ): Promise<Group | GroupWithMembers> {
+        // TODO: fix include member count
+        const existingGroup = await this.getGroupWithMembers(groupUuid, 10000);
+        if (existingGroup === undefined) {
             throw new NotFoundError(`No group found`);
         }
-        return {
-            uuid: updatedGroup.group_uuid,
-            name: updatedGroup.name,
-            createdAt: updatedGroup.created_at,
-            organizationUuid: updatedGroup.organization_uuid,
-        };
+
+        await this.database.transaction(async (trx) => {
+            let group;
+            if (update.name) {
+                group = await trx('groups')
+                    .update({ name: update.name })
+                    .where('group_uuid', groupUuid)
+                    .returning('*');
+            }
+            if (update.members !== undefined) {
+                const membersToAdd = differenceBy(
+                    update.members,
+                    existingGroup.members,
+                    'userUuid',
+                );
```

Comment:
We should make sure the values are unique. API users might send duplicate users.

```
test('check duplicate members ',() => {
    const membersToAdd = differenceBy(
        [{userUuid: 'a'}, {userUuid: 'a'}, {userUuid: 'b'}, {userUuid: 'b'}],
        [{userUuid: 'a'}, {userUuid: 'c'}],
        'userUuid',
    );
    expect(membersToAdd).toStrictEqual([{userUuid: 'b'}]);
})
```

Reply:
Right, that was why to do the .ignore(). Do you think it's better to thow if they are not unique? I don't have a good reason to go one way or the other so ignoring conflicts seems ok.